lms/djangoapps/certificates/api.py

```diff
@@ -221,7 +221,12 @@ def certificate_downloadable_status(student, course_key):
     Returns:
         Dict containing student passed status also download url, uuid for cert if available
     """
-    current_status = certificate_status_for_student(student, course_key)
+    from lms.djangoapps.courseware.courses import get_course_by_id
+
+    if course_key and not get_course_by_id(course_key).may_certify():
+        current_status = {'status': CertificateStatuses.date_unavailable}
```

Comment:
I'm not 100% clear on the hierarchy of cert statuses, so I may be wrong about this, but I thought that we only wanted the `date_unavailable` status to apply to verified learners who've passed the course but can't get their cert because the cert available date is in the future. As I understand it, this code will return a `date_unavailable` status for all learners, even audit track or blacklisted ones, until the cert available date has passed.